### ISSUE TITLE
Rule-specific eslint-enable comments don't work after global eslint-disable comments

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#9215]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
rules:
    semi: error
```

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint-disable */
/* eslint-enable semi */

foo
```

**What did you expect to happen?**

I expected a `semi` error to be reported on line 4, because the `semi` rule was reenabled on line 2.

**What actually happened? Please include the actual, raw output from ESLint.**

No error was reported.

I found this issue while refactoring the `eslint-disable` logic. I have a fix in progress.
